### ISSUE TITLE
[v3] Implement new Concept Exercise: mutation

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#852]
This issue describes how to implement the `mutation` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Mutation in F#.

## Learning objectives

- Know how mutation works in F#.
- Know how to define a variable (mutable binding).
- Know how to work with reference cells.
- Know how to update mutable values.
- Know when to use mutability.

## Out of scope

- How the compiler sometimes translates functional code to mutation.

## Concepts

The Concepts this exercise unlocks are:

- `mutation`: know how mutation works in F#; know how to define a variable (mutable binding); know how to work with reference cells; know how to update mutable values; know when to use mutability.

## Prerequisites

This exercise's prerequisites Concepts are:

- `generic-types`: know what a generic type is.
- `records`: know how to work with records.

Any data types used in this exercise (e.g. `strings` or `integers`) should also be added as prerequisites.

### Hints

- [Variables][variables]: how to define variables.
- [Reference cells][reference-cells]: how to work with reference cells.

### After

- [Variables][variables]: how to define variables.
- [Reference cells][reference-cells]: how to work with reference cells.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/types/map.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[variables]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/values/#mutable-variables
[reference-cells]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/reference-cells
